packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -90,28 +85,16 @@ export default function UnsavedInnerBlocks( {
 		}
 	);
 
-	const { isSaving, hasResolvedDraftNavigationMenus } = useSelect(
+	const { isSaving } = useSelect(
 		( select ) => {
 			if ( isDisabled ) {
 				return EMPTY_OBJECT;
 			}
 
-			const {
-				getEntityRecords,
-				hasFinishedResolution,
-				isSavingEntityRecord,
-			} = select( coreStore );
+			const { isSavingEntityRecord } = select( coreStore );
 
 			return {
 				isSaving: isSavingEntityRecord( 'postType', 'wp_navigation' ),
-				draftNavigationMenus: getEntityRecords(
-					// This is needed so that hasResolvedDraftNavigationMenus gives the correct status.
-					...DRAFT_MENU_PARAMS
-				),
-				hasResolvedDraftNavigationMenus: hasFinishedResolution(
```

Comment:
How about we leave this in but update to this:

```
hasResolvedAllNavigationMenus: hasFinishedResolution(
	'getEntityRecords',
	[
		'postType',
		'wp_navigation',
		{ 
			per_page: -1, 
			status: [ 'publish', 'draft' ] 
		}
	]
)
```

Why? Because we need to know that _all_ menus have resolved before we allowed more menus to be created. Therefore we shouldn't rely on configuration elsewhere in the codebase for this resolution status. Rather we should simply do it directly here.

This will ensure that whatever happens elsewhere in the code in the future this check will _always_ wait for _all_ menus to be resolved before proceeding.

Reply:
Is your concern that we might one day change `hasResolvedNavigationMenus` to only show published ones?

Reply:
Yes. Or we might not have that at all.

It's much safer to write a dedicated resolver selector to test for the exact statuses we want. Whilst more verbose the consequences of this breaking are quite severe (multiple duplicate menus) so it's worth keeping everything transparent.

Look how we've all lost context of why this whole thing even existed. Having a dedicated selector and a clear comment in the code explaining why it's there would ensure that doesn't happen again.

Reply:
I updated this to use the new selector as you suggested.